airflow/providers/delta/sharing/hooks/delta_sharing.py

```diff
@@ -0,0 +1,286 @@
+        headers['Authorization'] = f'Bearer {token}'
+
+        request_func: Any
+        if http_method == 'GET':
+            request_func = requests.get
+        elif http_method == 'HEAD':
+            request_func = requests.head
+        elif http_method == 'POST':
+            request_func = requests.post
+        else:
+            raise AirflowException('Unexpected HTTP Method: ' + http_method)
+
+        try:
+            for attempt in self._get_retry_object():
+                with attempt:
+                    response = request_func(
+                        url,
+                        json=json if http_method == 'POST' else None,
+                        params=json if http_method in ('GET', 'HEAD') else None,
+                        headers=headers,
+                        timeout=self.timeout_seconds,
+                    )
+                    print(f"HTTP Status: {response.status_code}")
+                    response.raise_for_status()
+                    return response
+        except RetryError:
+            raise AirflowException(
+                f'API requests to Delta Sharing failed {self.retry_limit} times. Giving up.'
+            )
+        except requests_exceptions.HTTPError as e:
+            raise AirflowException(f'Response: {e.response.content}, Status Code: {e.response.status_code}')
+
+    @staticmethod
+    def _retryable_error(exception: BaseException) -> bool:
+        if not isinstance(exception, requests_exceptions.RequestException):
+            return False
+        return isinstance(exception, (requests_exceptions.ConnectionError, requests_exceptions.Timeout)) or (
+            exception.response is not None
+            and (exception.response.status_code >= 500 or exception.response.status_code == 429)
+        )
+
+    @staticmethod
+    def _extract_delta_sharing_version(response, item_name: str) -> int:
+        version = None
+        for k, v in response.headers.lower_items():
+            if k == 'delta-table-version':
+                version = v
+        if version is None:
+            raise AirflowException(
+                f"No delta-table-version header in response from Delta Sharing server for item {item_name}"
+            )
+
+        return int(version)
+
+    def get_table_version(self, share: str, schema: str, table: str) -> int:
+        """
+        Returns a version of the Delta Sharing table
+        :param share: name of the share in which check will be performed.
+        :param schema: name of the schema (database) in which check will be performed.
+        :param table: name of the table to check.
+        :return: version of the given Delta Sharing table
+        """
+        response = self._do_api_call(
+            f"shares/{share}/schemas/{schema}/tables/{table}",
+            http_method='HEAD',
+        )
+        return self._extract_delta_sharing_version(response, f"{share}.{schema}.{table}")
+
+    def query_table(
```

Comment:
Perfect example to use this instead https://github.com/delta-io/delta-sharing/blob/2b0c440ed9ef2da496a7cfd590f630e873f18a03/python/delta_sharing/rest_client.py#L242 and I can totally see how it's valuable to add the version to the `ListFilesInTableResponse` class to avoid race conditions... https://github.com/delta-io/delta-sharing/blob/2b0c440ed9ef2da496a7cfd590f630e873f18a03/python/delta_sharing/rest_client.py#L77

Reply:
List files could have more overhead on the server side - that's why it's documented separately in the Delta Sharing protocol.